modules/webgl/src/adapter/resources/webgl-transform-feedback.ts

```diff
@@ -0,0 +1,253 @@
+import type {Device, ShaderLayout, TransformFeedbackProps} from '@luma.gl/core';
+import {log, isObjectEmpty, TransformFeedback, Buffer} from '@luma.gl/core';
+import {GL} from '@luma.gl/constants';
+import {WebGLDevice} from '../webgl-device';
+import {WEBGLBuffer, isWebGL2} from '../..';
+
+/** For bindRange */
+type BufferRange = {
+  buffer: WEBGLBuffer;
+  byteOffset?: number;
+  byteLength?: number;
+};
+
+export class WEBGLTransformFeedback extends TransformFeedback {
+  get [Symbol.toStringTag](): string {
+    return 'TransformFeedback';
+  }
+
+  readonly device: WebGLDevice;
+  readonly gl2: WebGL2RenderingContext;
+  readonly handle: WebGLTransformFeedback;
+
+  /**
+   * NOTE: The Model already has this information while drawing, but
+   * TransformFeedback currently needs it internally, to look up
+   * varying information outside of a draw() call.
+   */
+  readonly layout: ShaderLayout;
+  buffers: Record<string, BufferRange> = {};
+  unusedBuffers: Record<string, Buffer> = {};
+  // NOTE: The `bindOnUse` flag is a major workaround:
```

Comment:
Nit. I prefer `/** */` style comments for (all) fields as they are used by vscode when hovering.